fuel-tests/tests/contract.rs

```diff
@@ -1,29 +1,64 @@
-use fuel_core::{
-    database::Database,
-    service::{Config, FuelService},
-};
-use fuel_gql_client::client::FuelClient;
-use fuel_storage::Storage;
-use fuel_vm::prelude::Contract as FuelVmContract;
+use crate::helpers::{TestContext, TestSetupBuilder};
+use fuel_vm::prelude::*;
 
 #[tokio::test]
-async fn contract() {
-    // setup test data in the node
-    let contract = FuelVmContract::default();
-    let id = fuel_types::ContractId::new(Default::default());
-
-    let mut db = Database::default();
-    Storage::<fuel_types::ContractId, FuelVmContract>::insert(&mut db, &id, &contract).unwrap();
-    // setup server & client
-    let srv = FuelService::from_database(db, Config::local_node())
+async fn test_contract_salt() {
+    let mut test_builder = TestSetupBuilder::new(2322);
+    let (_, contract_id) = test_builder.setup_contract(vec![], None);
+
+    // spin up node
+    let TestContext { client, .. } = test_builder.finalize().await;
+
+    let contract = client
+        .contract(format!("{:#x}", contract_id).as_str())
         .await
         .unwrap();
-    let client = FuelClient::from(srv.bound_address);
 
-    // run test
-    let contract = client
-        .contract(format!("{:#x}", id).as_str())
+    // Check that salt is 0x Hex prefixed
+    let salt = contract.unwrap().salt;
+    assert_eq!("0x", &salt.to_string()[..2]);
+}
+
+#[tokio::test]
+async fn test_contract_balance_default() {
+    let mut test_builder = TestSetupBuilder::new(2322);
+    let (_, contract_id) =
+        test_builder.setup_contract(vec![], Some(vec![(AssetId::new([1u8; 32]), 0)]));
+
+    // spin up node
+    let TestContext { client, .. } = test_builder.finalize().await;
+
+    let asset_id = AssetId::new([1u8; 32]);
+
+    let balance = client
+        .contract_balance(
+            format!("{:#x}", contract_id).as_str(),
+            Some(format!("{:#x}", asset_id).as_str()),
+        )
         .await
         .unwrap();
-    assert!(contract.is_some());
+
+    assert_eq!(balance, 0); // Ensures default value is working properly
+}
+
+#[tokio::test]
+async fn test_contract_balance() {
+    let mut test_builder = TestSetupBuilder::new(2322);
+    let (_, contract_id) =
+        test_builder.setup_contract(vec![], Some(vec![(AssetId::new([1u8; 32]), 500)]));
```

Comment:
move magic number into constant variable at start of the test.

Reply:
Moved it to a const at the beginning of the file and then pointed all the test_build initializations to it

Reply:
oh i meant the balance var `500`

Reply:
I did slightly more than that, instead of making it a const I just merged it with the previous test (which did the same thing except with a balance of 0) to instead have on test which tries with various different potential balances